javascript/ql/src/semmle/javascript/StandardLibrary.qll

```diff
@@ -209,6 +209,21 @@ private class PromiseFlowStep extends DataFlow::AdditionalFlowStep {
   }
 }
 
+/**
+ * A data flow edge from the exceptional return of the promise executor to the promise catch handler.
+ */
+class PromiseExceptionalStep extends DataFlow::AdditionalFlowStep {
```

Comment:
Is this step really feature complete enough yet? Perhaps we should just keep it private, and make a note of the features that we are aware that it is missing:

- support for flow of exceptions to the catch block of an `await`
- support for flow of exceptions to the first catch-handler in a chain of promise-steps: `new Promise(res, rej).then(..).then(..).catch(e => FLOWS_HERE).then(...).catch(e => BUT_NOT_HERE)`

Also, since this is a default taint step for all `TaintTracking::Configuration`, I want to make sure that we have done an evaluation with `--suite security`.

Reply:
I put in a comment about what is missing.  
I'll maybe return to it at some and add more flow edges. 

I'll get an evaluation going.